task/host-sp-comms/src/main.rs

```diff
@@ -600,12 +618,23 @@ impl ServerImpl {
                 })
             }
             HostToSp::GetMacAddresses => {
-                // TODO where do we get host MAC addrs?
-                Some(SpToHost::MacAddresses {
-                    base: [0; 6],
-                    count: 1,
-                    stride: 1,
-                })
+                let block = self.net.get_leftover_mac_addresses();
+                let response = if block.count.get() > 0 {
+                    let count =
+                        u16::min(block.count.get(), NUM_HOST_MAC_ADDRESSES);
+                    SpToHost::MacAddresses {
+                        base: block.base_mac,
+                        count,
+                        stride: block.stride,
+                    }
+                } else {
+                    SpToHost::MacAddresses {
```

Comment:
Is it important to return zeros for `base` and `stride` here?  Otherwise, I'm not sure whether this block is needed: the host should notice `count: 0` either way (and freak out accordingly).

Reply:
I think I'd classify this as "nice to have" and not "critical" - I agree the host _should_ notice `count: 0` either way, but it seems safer in some sense to give it all zeroes in this case: even if the host is buggy and assumes at least one address exists, that address will be nonsense.

Reply:
Okay, seems reasonable